src/group/managed_group/mod.rs

```diff
@@ -186,54 +425,223 @@ impl ManagedGroup {
         &mut self,
         credential_bundle: &CredentialBundle,
     ) -> Result<Vec<MLSMessage>, ManagedGroupError> {
-        unimplemented!()
+        // Create new KeyPackageBundle for own leaf
+        let tree = self.group.tree();
+        let existing_key_package = tree.get_own_key_package_ref();
+        let key_package_bundle = KeyPackageBundle::from_rekeyed_key_package(existing_key_package);
+        drop(tree);
+
+        // Create UpdateProposal
+        let mut plaintext_messages = vec![self.group.create_update_proposal(
+            &self.aad,
+            credential_bundle,
+            key_package_bundle.get_key_package().clone(),
+        )];
+
+        // Include pending proposals into Commit
+        let mut messages_to_commit = self.pending_proposals.clone();
```

Comment:
Not sure how large the `pending_proposals` will get. I just wonder if we couldn't avoid cloning it every time by only using references (we only pass in references to the group function anyway).

Reply:
Agreed, I creates #234 for that.